Backend/controllers/PatientsController.js

```diff
@@ -100,13 +102,46 @@ exports.post_patient_data = async (req, res) => {
     patient._id= mongoose.Types.ObjectId();
 
     try {
-        await patient.save();
-        return res.send(patient);
+        return link_patient_with_user(patient,res)
```

Comment:
> It would be nice to have link_patient_with_user and then return user and have this function return the response.
Source: Clean Code page 44: Have No Side Effects :)